src/main/java/net/modfest/scatteredshards/networking/ScatteredShardsNetworking.java

```diff
@@ -1,173 +1,318 @@
 package net.modfest.scatteredshards.networking;
 
-import com.google.common.collect.Multimap;
-import com.google.common.collect.MultimapBuilder;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
 import me.lucko.fabric.api.permissions.v0.Permissions;
 import net.fabricmc.api.EnvType;
 import net.fabricmc.api.Environment;
 import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;
-import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
 import net.fabricmc.fabric.api.networking.v1.PacketByteBufs;
+import net.fabricmc.fabric.api.networking.v1.PacketSender;
 import net.fabricmc.fabric.api.networking.v1.ServerPlayConnectionEvents;
 import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
 import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayNetworkHandler;
+import net.minecraft.nbt.NbtCompound;
 import net.minecraft.network.PacketByteBuf;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayNetworkHandler;
 import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.util.Identifier;
 import net.modfest.scatteredshards.ScatteredShards;
 import net.modfest.scatteredshards.api.ScatteredShardsAPI;
-import net.modfest.scatteredshards.api.impl.ScatteredShardsAPIImpl;
-import net.modfest.scatteredshards.client.ScatteredShardsClient;
-import net.modfest.scatteredshards.component.ScatteredShardsComponents;
+import net.modfest.scatteredshards.api.ShardLibrary;
+import net.modfest.scatteredshards.api.impl.ShardCollectionPersistentState;
+import net.modfest.scatteredshards.api.impl.ShardLibraryPersistentState;
 import net.modfest.scatteredshards.api.shard.Shard;
 import net.modfest.scatteredshards.api.shard.ShardType;
-import net.modfest.scatteredshards.load.ShardSetLoader;
-import net.modfest.scatteredshards.load.ShardTypeLoader;
-
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Map;
+import net.modfest.scatteredshards.client.ScatteredShardsClient;
 
 public class ScatteredShardsNetworking {
```

Comment:
I think these packet classes should be in different files under the networking package